src/dune_util/event_queue.mli

```diff
@@ -0,0 +1,28 @@
+(** Stores an association between monotonically increasing integer sequence
+    numbers and events. There is a limit to the number of events that can be
+    stored, and once this limit is reached, adding a new event will cause the
+    oldest event to be removed. *)
+type 'a t
+
+(** Create a new event queue *)
```

Comment:
I'd move this module `dune_rpc_impl`. It seems like an implementation detail of rpc, so I don't see a need to make it available to anyone else